pkg/internal/common/jsoncanonicalizer/jsoncanonicalizer.go

```diff
@@ -255,20 +256,35 @@ func Transform(jsonData []byte) (result []byte, e error) {
 	}
 
 	parseArray = func() string {
-		var arrayData strings.Builder
-		arrayData.WriteByte('[')
+		var elements []string
+
 		var next bool = false
 		for globalError == nil && testNextNonWhiteSpaceChar() != ']' {
 			if next {
 				scanFor(',')
-				arrayData.WriteByte(',')
 			} else {
 				next = true
 			}
-			arrayData.WriteString(parseElement())
+			elements = append(elements, parseElement())
 		}
 		scan()
-		arrayData.WriteByte(']')
+
+		// sort the elements so we enforce an order
```

Comment:
Can we also comment on the sort strategy?

I assume we are following the same rules as for the property strings in objects? https://tools.ietf.org/html/draft-rundgren-json-canonicalization-scheme-17#section-3.2.3

Reply:
also - should update the trustbloc spec with some information on comparisons (as a followup).